hal/src/timer.rs

```diff
@@ -47,11 +47,26 @@ where
         T: Into<Hertz>,
     {
         let timeout = timeout.into();
-        let params = clock::ClockParams::new(self.freq, timeout);
-        let divider = params.divider;
+        let ticks: u32 = self.freq.0/timeout.0.saturating_sub(1).max(1);
```

Comment:
@sajattack I don't think we need the `saturating_sub` here anymore. It breaks 2Hz